assets/javascripts/kitten/components/cards/triptych.js

```diff
@@ -0,0 +1,158 @@
+import React, { Component } from 'react'
+import Radium, { StyleRoot } from 'radium'
+import PropTypes from 'prop-types'
+import { Grid, GridCol as GridColBase } from 'kitten/components/grid/grid'
+import { Marger as MargerBase } from 'kitten/components/layout/marger'
+import { VerticalStroke } from 'kitten/components/layout/vertical-stroke'
+import { ScreenConfig } from 'kitten/constants/screen-config'
+import { mediaQueries } from 'kitten/hoc/media-queries'
+import { debounce } from 'kitten/helpers/utils/debounce'
+import { GUTTER } from 'kitten/constants/grid-config'
+
+const Marger = Radium(MargerBase)
+const GridCol = Radium(GridColBase)
+
+class TriptychBase extends Component {
+  constructor(props) {
+    super(props)
+
+    this.state = { secondCardComputedTopMargin: null }
+  }
+
+  updateSecondCardMargin = () => {
+    this.setState((state, props) => {
+      const { viewportIsTabletOrLess, viewportIsSOrLess } = props
+      const isTablet = viewportIsTabletOrLess && !viewportIsSOrLess
```

Comment:
Non je ne peux pas. La subtilité ici c'est que j'utilise le `props` de la méthode `this.setState`.
En utilisant une méthode de class, je ne récupère ni `viewportIsTabletOrLess` ni `viewportIsSOrLess`. Ma valeur dans `isTablet` est donc erroné au chargement du composant.

Reply:
Oui car j'utilise `updateSecondCardMargin` dans `componentDidMount` et la valeur de `this.props` est setté de manière asynchrone CF : [state-updates-may-be-asynchronous](https://reactjs.org/docs/state-and-lifecycle.html#state-updates-may-be-asynchronous)
> Because `this.props` and `this.state` may be updated asynchronously, you should not rely on their values for calculating the next state.